llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1932,8 +2000,15 @@ void AsmPrinter::emitFunctionBody() {
 
   // Emit section containing BB address offsets and their metadata, when
   // BB labels are requested for this function. Skip empty functions.
-  if (MF->hasBBLabels() && HasAnyRealCode)
+  bool HasBBLabels = MF->hasBBLabels();
+  if (HasBBLabels && HasAnyRealCode)
```

Comment:
This is equivalent to:
```
if (HasAnyRealCode) {
  if (MF->hasBBLabels())
    emitBBAddrMapSection(*MF)
  else if (PgoAnalysisMapFeatures.getBits() != 0)
    \\ report warning
 }
```

Reply:
Nice, I hadn't thought to swap the condition.